src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -871,35 +873,46 @@ private boolean circulationNoteChanged(CirculationNote newNote, CirculationNote
   /**
    * Fetches bound-with parts and referenced holdingsRecords and Instances
    * and builds a list of bound-with titles that is set on the provided Item.
-   * NOTE: This method will mutate the argument item (adding data to it)
+   * The method will mutate the argument 'item'.
    *
    * @param item The Item to set bound-with titles on
    * @param boundWithPartsClient Client for retrieving bound-with parts from storage
-   * @param holdingsClient Client for retrieving holdings from storage
-   * @param instancesClient Client for retrieving Instance from storage
-   * @return Future with null Response (Using Response return type in order to
-   * add this future to a list of other futures returning Response)
-   */
+  */
   private CompletableFuture<Response> setBoundWithTitlesOnItem(
     Item item,
     CollectionResourceClient boundWithPartsClient,
-    CollectionResourceClient holdingsClient,
-    CollectionResourceClient instancesClient
+    RoutingContext routingContext
   ) {
-    return getBoundWithPartsForItemFuture( item, boundWithPartsClient ).thenCompose( partsResponse -> {
-      JsonArray boundWithParts = partsResponse.getJson().getJsonArray("boundWithParts" );
-      if ( boundWithParts.isEmpty() ) {
-        item.withIsBoundWith( false );
-        return CompletableFuture.completedFuture( null );
-      } else {
-        return fetchHoldingsByForeignKeys( boundWithParts, holdingsClient ).thenCompose(
-          holdingsRecords -> fetchInstancesByForeignKeys( holdingsRecords, instancesClient ).thenCompose(
-            instances -> {
-              JsonArray boundWithTitles = buildBoundWithTitlesArray( holdingsRecords, instances );
-              item.withBoundWithTitles( boundWithTitles ).withIsBoundWith( true );
-              return CompletableFuture.completedFuture( null );
-            } ) );
-      }});
+    return getBoundWithPartsForItemFuture( item, boundWithPartsClient )
```

Comment:
I think it could be good to start extracting some of this behaviour out of the items class and into separate classes, as the class has gotten rather large (over 1000 lines)

Reply:
All right.

Reply:
How was this resolved?